schema-registry/authorizer/authorization-agent/src/main/java/com/hortonworks/registries/schemaregistry/authorizer/agent/DefaultAuthorizationAgent.java

```diff
@@ -0,0 +1,515 @@
+                                             String schemaMetadataName)
+            throws AuthorizationException, SchemaNotFoundException {
+
+        SchemaMetadataInfo smi = schemaRegistry.getSchemaMetadataInfo(schemaMetadataName);
+        if(smi == null) {
+            throw new SchemaNotFoundException("No SchemaMetadata exists with key: " + schemaMetadataName);
+        }
+
+        authorizeSerDes(userAndGroups, AccessType.READ);
+        authorizeSchemaMetadata(userAndGroups, smi.getSchemaMetadata(), AccessType.UPDATE);
+    }
+
+    @Override
+    public void authorizeMergeSchemaVersion(UserAndGroups userAndGroups,
+                                            ISchemaRegistry schemaRegistry,
+                                            Long versionId)
+            throws AuthorizationException, SchemaNotFoundException {
+
+        SchemaVersionInfo schemaVersionInfo = schemaRegistry.getSchemaVersionInfo(new SchemaIdVersion(versionId));
+        SchemaMetadata sM = schemaRegistry.getSchemaMetadataInfo(schemaVersionInfo.getSchemaMetadataId()).getSchemaMetadata();
+        String sGroup = sM.getSchemaGroup();
+        String sName = sM.getName();
+
+        String branch = getPrimaryBranch(schemaRegistry.getSchemaBranchesForVersion(versionId));
+
+        Authorizer.SchemaVersionResource schemaVersionResource =
+                new Authorizer.SchemaVersionResource(sGroup, sName, branch);
+        authorize(schemaVersionResource, AccessType.READ, userAndGroups);
+
+        Authorizer.SchemaVersionResource schemaVersionToCreate =
+                new Authorizer.SchemaVersionResource(sGroup, sName, SchemaBranch.MASTER_BRANCH);
+        authorize(schemaVersionToCreate, AccessType.CREATE, userAndGroups);
+    }
+
+    ///////////////////// ConfluenceCompatible APIs /////////////////////
+
+    @Override
+    public Collection<SchemaVersionInfo> authorizeGetAllVersions(UserAndGroups userAndGroups,
+                                                                 ISchemaRegistry schemaRegistry,
+                                                                 Collection<SchemaVersionInfo> versions)
+            throws SchemaNotFoundException {
+
+        return authorizeGetEntities(userAndGroups, versions, schemaVersionInfo -> {
+            SchemaMetadata sM =
+                    schemaRegistry.getSchemaMetadataInfo(schemaVersionInfo.getSchemaMetadataId())
+                            .getSchemaMetadata();
+            String sGroup = sM.getSchemaGroup();
+            String sName = sM.getName();
+            String sBranch = getPrimaryBranch(schemaRegistry.getSchemaBranchesForVersion(schemaVersionInfo.getId()));
+
+            return new Authorizer.SchemaVersionResource(sGroup, sName, sBranch);
+        });
+
+    }
+
+    ////////////////////////////////////////////////////////////////////////
+
+    public static class AlreadyConfiguredException extends RuntimeException {
```

Comment:
Can you create an "exception" module and move this class there ?

Reply:
We can do that. But why do we need to do this? This exception is used only for DefaultAuhtorizationAgent.